# Ejemplo practico energia solar termica para suministro de ACS para vivienda unifamiliar

Se quiere desarrollar un proyecto de instalación solar térmica de baja temperatura para suministro de agua caliente sanitaria (ACS) en una vivienda unifamiliar de tres dormitorios, de nueva construcción. La vivienda está situada en Alcalá de Henares (Madrid).


La instalación está compuesta por el sistema de captación solar, con captador solar plano, situado en el tejado del edificio y un sistema de intercambio y acumulación conjunto (interacumulador) y centralizado, único en la vivienda. El sistema de aporte de energía convencional auxiliar es de gas natural.

Los datos del proyecto son los siguientes:

* Localidad: Alcalá de Henares (Madrid).
* Latitud: 40,5 N.
* Tipo de edificio: Vivienda unifamiliar.
* Tipo de integración de captadores: integración arquitectónica.
* Sistema de energía auxiliar: gas natural.
* Orientación de campo de captadores: sur.
* Inclinación de campo de captadores: 35°.
* Condiciones del entorno: sin sombras posibles.

El esquema del sistema de suministro de ACS se ilustra en la siguiente figura:

![Esquema_ACS_unifam](Esquema-solartermica-unifamiliar.png)

Para comenzar, se enumerarán los puntos que como justificación del cumplimiento de la exigencia el CTE DB-HE 4 2013 indica que deben constar en el proyecto, para a continuación ir dando respuesta a cada uno de ellos:

1. Zona climática.
2. Demanda anual de ACS.
3. Contribución solar mínima exigida.
4. Características y dimensionado de la instalación solar térmica.
5. Contribución solar anual alcanzada.
6. Plan de vigilancia y plan de mantenimiento.

## 1. Zona Climática

Para resolver este punto observamos el mapa de zonas climáticas de España a continuación:

![Zonas_climaticas_España](Zonas-climaticas-ESP.jpg)

Alcalá de Henares se encuentra muy próximo a Madrid. La zona climática que le corresponde es la IV, la cual cuenta con una buena radiación solar, no tan buena como la zona V (sur de España y Canarias) pero mucho mejor que la I (norte de España).

## 2. Demanda Anual de Agua Caliente Sanitaria (ACS)

Para una vivienda unifamiliar de tres habitaciones se considera que el número de habitantes es cuatro (4). La tabla de referencia de demanda de ACS diaria del según el CTE DB-HE4 2013 nos da una idea de cuantos litros de agua caliente demanda una persona según el tipo de inmueble.
![Tabla-demanda-ACS-dia](Tabla-demanda-ACS-dia.jpg)

En base a esta tabla tenemos que una persona en una vivienda familiar consume 28 litros de agua caliente por día.

In [1]:
Consumo_diario = 28 * 4
print("La vivienda requiere {} litros de ACS diariamente".format(Consumo_diario))

La vivienda requiere 112 litros de ACS diariamente


En total, la vivienda unifamiliar de cuatro personas demandará por día 112 litros o 0,112 m3 de ACS 

La demanda anual la obtenemos multiplicando la demanda diaria por los 365 días que contiene un año:

In [2]:
Dem_ACS_anual = Consumo_diario * 365 / 1000 #Convertido a m3 dividiendo litros/1000
print("La demanda anual de ACS de la vivienda unifamiliar es {:.2f} m3".format(Dem_ACS_anual))

La demanda anual de ACS de la vivienda unifamiliar es 40.88 m3


## 3. Contribución solar mínima exigida

De acuerdo al CTE, la contribución solar mínima exigida para suministrar ACS en un edificio depende de la zona climática donde se encuentre y de la demanda diaria, como se resume en la siguiente tabla:

![contribucion-solar-minima-ACS](contribucion-solar-minima-ACS.jpg)

Dado que esta vivienda se encuentra en la zona climática IV y que la demanda diaria de ACS es de 112 litros, mucho menos de 5 mil litros, la contribución solar mínima es 50% de la demanda energética (DE) total. Para cuantificar en nuestro caso la contribución solar de energía necesaria debemos estimar la demanda energética total. Dicha demanda es afectada por la temperatura de agua de red, que mes a mes varía, así como la cantidad de días de cada mes. Por lo tanto, la demanda energética hay que calcularla por mes

La demanda energética mensual para un mes *i* viene dada por la siguiente fórmula:

$ DE_{mes,i} = D_{dia}.N_{días,i}.C_p.\rho.(T_{ACS}-T_{red,i}) $

Donde:
* $D_{dia}$: Demanda diaria de ACS ($m^3$)
* $N_{días,i}$: Número de días del mes *i*
* $C_p$: Calor específico del agua (1 Kcal/Kg.°C)
* $\rho$: Densidad del agua, típicamente 1.000 Kg/$m^3$
* $T_{ACS}$: Temperatura de agua caliente, 60 °C
* $T_{red,i}$: Temperatura promedio de agua de red en el mes *i*

La Demanda energética obtenida de la anterior ecuación viene expresada en Kcal, para transformarla a KWh es neceario escarlarla por $1,16.10^{-3} KWh/Kcal$

La siguiente tabla nos ilustra las temperaturas de agua de red mensuales, $T_{red}$, para cada capital provincial de España.

![Temperaturas-agua_red-Esp-mes](Temperaturas-agua_red-Esp-mes.png)

Alcalá de Henares está muy próxima a Madrid, pero a una elevación de 594 m.s.n.m., 61 metros más bajo que Madrid. Bien pudiéramos despreciar el efecto de esa pequeña diferencia de altitud, pero podemos aplicar la fórmula de ajuste de temperaturas de la guía ASIT en función de la diferencia de altitud con respecto a una referencia cercana conocida y el mes de año:

$T_{red,i} = T_{redCP,i} - B.\Delta Z$

Donde:

* $T_{redCP,i}$: Temperatura por mes de agua de red de la capital provincial, valor conocido
* B: Constante equivalente a 0,0066 para los meses de octubre a marzo y 0,0033 para los meses de abril a septiembre.
* $\Delta Z$: Diferencia de altura entre la nueva localidad y la capital provincial.

Procedemos a calcular la temperatura de agua de red aproximada para Alcalá de Henares tomando en cuenta su diferencia de altitud con respecto a Madrid y la constante B para los distintos meses del año:

In [4]:
meses = ['ENE','FEB','MAR','ABR','MAY','JUN','JUL','AGO','SEP','OCT','NOV','DIC']
dias = [31,28,31,30,31,30,31,31,30,31,30,31]
Tred_MAD = [8,8,10,12,14,17,20,19,17,13,10,8]
DZ = 594 - 655
Tred = []
for i in range(12):
    if i in [0,1,2,9,10,11]: #En Python el primer indice es 0, correspondiente a enero, dic. es 11
        Tred.append(round(Tred_MAD[i] - 0.0066*DZ, 1))
    else:
        Tred.append(round(Tred_MAD[i] - 0.0033*DZ, 1))
Tred

[8.4, 8.4, 10.4, 12.2, 14.2, 17.2, 20.2, 19.2, 17.2, 13.4, 10.4, 8.4]

Una vez tenemos conocida la temperatura de red local, calculamos la demanda energética requerida para calentar el ACS a 60°C.

In [6]:
DE_mes = []
for i in range(12):
    DE_mes.append(round(0.00116*(Consumo_diario/1000)*dias[i]*1*1000*(60-Tred[i]), 2))
DE_mes

[207.82,
 187.71,
 199.76,
 186.31,
 184.46,
 166.82,
 160.3,
 164.32,
 166.82,
 187.68,
 193.32,
 207.82]

Para facilitar nuestra lectura y almacenamiento de la información, la cargamos a una tabla Pandas:

In [5]:
import pandas as pd

In [7]:
df = pd.DataFrame(zip(meses,dias,Tred,DE_mes),columns=['MES','DIAS','T_red','DE_mes'])
df.head(12)

MES  DIAS  T_red  DE_mes
0   ENE    31    8.4  207.82
1   FEB    28    8.4  187.71
2   MAR    31   10.4  199.76
3   ABR    30   12.2  186.31
4   MAY    31   14.2  184.46
5   JUN    30   17.2  166.82
6   JUL    31   20.2  160.30
7   AGO    31   19.2  164.32
8   SEP    30   17.2  166.82
9   OCT    31   13.4  187.68
10  NOV    30   10.4  193.32
11  DIC    31    8.4  207.82

Como podemos observar, los meses de invierno son los que mayor demanda energética requieren para calentar el ACS. Julio, pleno verano, es el mes que menos energía demanda. La suma de las demandas mensuales nos da la demanda energética total en un año.

In [8]:
DE_anual = sum(df['DE_mes'])
print("Demanda energética anual: {} KWh".format(DE_anual))
Contrib_sol_min = 0.5 * DE_anual
print("Contribución solar mínima anual: {} KWh".format(Contrib_sol_min))

Demanda energética anual: 2213.14 KWh
Contribución solar mínima anual: 1106.57 KWh


Esta vivienda unifamiliar en Alcalá de Henares requiere 2.213,14 KWh para calnetar el agua caliente sanitaria durante un año. Dada la relativamente baja demanda diaria de ACS y la zona climática IV donde se ubica la edificación, la contribución solar mínima exigida tiene que ser el 50% de la demanda energética total en un año. Es decir que la contribución energética solar mínima deberá ser **1.106,6 KWh** por año.

## 4. Características y dimensionado de la instalación solar térmica.

Debemos ahora calcular el número de captadores y el volumen de acumulador necesarios para cumplir los objetivos de generación.

Un dato importante que necesitamos es la radiación solar que incide en la zona de interés. Además de esto importa la inclinación $\beta$ de los captadores. La radiación solar resultante viene dada por la siguiente ecuación:

$G_{di}(\beta) = k. G_{di}(0)$

$G_{di}(0)$ es la radiación horizontal (cero grados de inclinación) promedio mensual que se presenta en la localidad de interés. Dada la gran cercanía con Madrid, los datos conocidos para la capital española son aplicables para Alcalá de Henares.

Estos valores, KWh/m2, los almacenamos en la siguiente lista:

In [13]:
Gdi_MAD = [1.86,2.94,3.78,5.22,5.81,6.53,7.22,6.42,4.69,3.17,2.08,1.64]

El factor k de corrección por mes lo podemos descargar de la tabla de factores k elaborada por Censolar para distntas inclinaciones a una latitud de 40°N:
![factorK-correc-radiac-sol-Censolar](factorK-correc-radiac-sol-Censolar.jpg)

Los factores k que nos corresponden, latitud 40 e inclinación 35, son:

In [11]:
k_35 = [1.37,1.28,1.17,1.06,.98,.95,.98,1.07,1.21,1.37,1.47,1.45]

Las radiaciones solares incidentes con páneles a 35 grados de inclinación mirando hacia el sur por mes, en KWh/m2 son: 

In [14]:
Gdi_35 = []
for i in range(12):
    Gdi_35.append(round(Gdi_MAD[i]*k_35[i], 2))
Gdi_35

[2.55, 3.76, 4.42, 5.53, 5.69, 6.2, 7.08, 6.87, 5.67, 4.34, 3.06, 2.38]

La energía solar producida en un mes, $EI_{mes}$, por la radiación solar incidente a la inclinación determinada se obtiene multiplicando la radiación mensual promedio inclinada, $G_{di}(\beta)$ por el número de días del mes.

$EI_{mes} = G_{di}(\beta).N_{días,mes}$

In [15]:
EI_mes = []
for i in range(12):
    EI_mes.append(round(Gdi_35[i]*dias[i], 2))
EI_mes

[79.05,
 105.28,
 137.02,
 165.9,
 176.39,
 186.0,
 219.48,
 212.97,
 170.1,
 134.54,
 91.8,
 73.78]

La estimación de número de captadores y volumen de acumulación lo estimamos aplicando el método *f-chart*, que permite calcular de manera empírica la fracción solar mensual o $f_{mes}$ en función de dos variables D1 y D2. 

$f_{mes} = 1,029.D_1 - 0,065.D_2 - 0,245.D_1^2 + 0,0018.D_2^2 + 0,0215.D_1^3$

Para estimar estas variables vamos a necesitar unos datos del fabricante de los paneles:

| Parámetro | Propiedad |
| ---------------- | ----------------- |
|Modelo de capador | Fargor Solaria 2.4|
|Superficie apertura| 2,4 m2 |
| Eficiencia óptica | 0,67 |
| Coef. glob.perd. | 3,14 W/m2.°C |
| MAI | 0,95 |
| Fcint | 0,95 |

$D_1$ representa la relación entre la energía absorbida y la demanda energética por mes.

$D_1 = {E_{absor} \over DE_{mes}} = {S_C.\eta_0.MAI.F_{CInt}.G_{dm}.N_{dias} \over DE_{mes}}$

$S_C$ corresponde a la superficie del captador, 2,4 m2 en este caso; $\eta_0$ es la eficiencia óptica de 0,67; el modificador de ángulo de incidencia (MAI) es 0,95; $F_{CInt}$ el factor de corrección del conjunto captador-intercambiador; $G_{dm}$ la irradiación solar diaria promedio para ese mes y $N_{dias}$ el número de días del mes.

In [16]:
D1_mes = []
for i in range(12):
    D1_mes.append(round(2.4*0.67*0.95*0.95*Gdi_35[i]*dias[i]/DE_mes[i], 2))
D1_mes

[0.55, 0.81, 1.0, 1.29, 1.39, 1.62, 1.99, 1.88, 1.48, 1.04, 0.69, 0.52]

D2 representa las pérdidas del captador con respecto a la demanda mensual. Viene dado por la siguiente ecuación:

$D_2 = {E_{perd} \over DE_{mes}} = {S_C.K_{global}.F_{CInt}.(100 - T_{amb}).FC_{acum}.FC_{ACS}.24.N_{dias} \over DE_{mes}}$

$K_{global}$ es el coeficiente global de pérdida, dato que provee el fabricante de los paneles. $T_{amb}$ corresponde a la temperatura ambiente promedio mensual de la localidad de estudio, información disponible enn portales meteorológicos. Tenemos las siguientes temperaturas promedio de Alcalá de Henares:

In [17]:
Tamb_mes = [6,8,11,13,18,23,28,26,21,15,11,7]

Con respecto al factor de corrección del acumulador, $FC_{acum}$, tenemos una fórmula empírica que lo relaciona con el volumen del acumulador solar $V_{acum}$:

$FC_{acum} = ({{V_{acum} / S_C} \over {75 l/m^2}})^{-1/4}$

A su vez, la proporción entre el volumen del acumulador y la superficie del captador debe estar entre 50 y 180 litros por m2.

$50 l/m^2 < \frac {V_{acum}} {S_C} < 180 l/m^2$

In [18]:
import math

In [20]:
Facum_min = round(math.pow(180/75,-0.25), 2)
Facum_max = round(math.pow(50/75,-0.25), 2)
print("Factor de correccion del acumulador entre {} y {}".format(Facum_min, Facum_max))

Factor de correccion del acumulador entre 0.8 y 1.11


Tenemos que el factor de corrección del acumulador está entre valores de 0,8 y 1,11. En paralelo, si despejamos la fórmula de $\frac {V_{acum}} {S_C}$ limitada por los valores de 50 y 180 l/m2 podemos obtener un rango de volumenes de acumulador:

In [22]:
Vacum_max = 180* 2.4
Vacum_min = 50* 2.4
print("Volumen de acumulador sugerido entre {} y {} litros".format(Vacum_min, Vacum_max))

Volumen de acumulador sugerido entre 120.0 y 432.0 litros


Al principio de este caso determinamos que el consumo diario de ACS es de 112 litros. El pliego de condiciones técnicas del IDAE recomienda que el volumen de acumulación sea aproximado al del consumo diario. Por lo tanto, nos decidimos como volumen de acumulación $V_{acum}$ **120 litros o 0,12 m3**. En consecuencia, $F_{acum}$ a considerar será el máximo, de 1,11.

$FC_{ACS}$ es el factor de corrección que relaciona la temperatura mínima de agua caliente, 60°C en nuestro caso, la del agua de red y la ambiente. Se calcula de la siguiente manera, siendo variable por cada mes:

$FC_{ACS} = {11,6 + 1,18.T_{ACSmin} + 3,86.T_{red} - 2,32.T_{amb} \over 100 - T_{amb}}$

In [23]:
F_ACS = []
for i in range(12):
    F_ACS.append(round((11.6+1.18*60+3.86*Tred[i]-2.32*Tamb_mes[i])/(100-Tamb_mes[i]), 2)) 
F_ACS

[1.07, 1.05, 1.09, 1.14, 1.16, 1.24, 1.33, 1.3, 1.27, 1.17, 1.09, 1.06]

Ya contamos con todas las variables que nos permitan estimar D2 para cada mes. Importante asegurarse de que el coeficiente de pérdida $K_{global}$ y la demanda energética DE estén en la misma escala, en W o KWh:

In [25]:
D2_mes = []
for i in range(12):
    D2_mes.append(round(2.4*3.14*0.95*(100-Tamb_mes[i])*Facum_max*F_ACS[i]*24*dias[i]/(DE_mes[i]*1000), 2))
D2_mes

[2.86, 2.75, 2.87, 3.05, 3.05, 3.27, 3.53, 3.46, 3.44, 3.13, 2.87, 2.8]

Con estos datos ya podemos calcular la fracción solar mensual del método *f-chart*:

In [26]:
f_mes = []
for i in range(12):
    f_mes.append(round(1.029*D1_mes[i]-0.065*D2_mes[i]-0.245*math.pow(D1_mes[i],2)+0.0018*math.pow(D2_mes[i],2)+0.0215*math.pow(D1_mes[i],3), 4))
f_mes

[0.3242,
 0.519,
 0.6338,
 0.7844,
 0.8332,
 0.9221,
 1.0399,
 1.0081,
 0.8537,
 0.6435,
 0.4287,
 0.304]

Una vez que conocemos la fracción de cada mes, podemos estimar la energía solar útil aportada al agua multiplicando dicha fracción por la correspondiente demanda energética de cada mes:

$EU_{solar,mes} = f_{mes}.DE_{mes}$

In [32]:
EU_solar = []
for i in range(12):
    EU_solar.append(round(f_mes[i]*DE_mes[i],2))
EU_solar

[67.38,
 97.42,
 126.61,
 146.14,
 153.69,
 153.82,
 166.7,
 165.65,
 142.41,
 120.77,
 82.88,
 63.18]

Vamos a estimar que porcentaje o fracción solar anual, *f* representa el total de la energía solar útil con respecto a la demanda energética anual:
$f = \frac {\sum EU_{solar,mes}} {\sum DE_{mes}}$ 

In [33]:
f = sum(EU_solar) / sum(DE_mes)
print("Fracción de la demanda energética aportada por la energía solar: {:.1f}%".format(f*100))

Fracción de la demanda energética aportada por la energía solar: 67.2%


Como resultado tenemos que un panel de la descripción técnica dada, en esta vivienda de 3 habitaciones ubicada en Alcalá de Henares es capaz de proveer un **67%** de la energía demandada para calentar 112 litros diarios de ACS, la cual es almacenada en un acumulador de 120 litros. Excedemos el mínimo que la energía solar debe suplir para esta vivienda unifamiliar. 

## 5. Contribución solar anual alcanzada

In [35]:
print("Contribución solar anual: {} KWh".format(round(sum(EU_solar)),2))

Contribución solar anual: 1487 KWh


Al cubrir el 67% de la demanda energética anual estamos contribuyendo con **1487 KWh al año**.

Podemos resumir en una tabla los parámetros obtenidos en este caso de estudio:

In [36]:
columnas = ['MES','DIAS','T_RED','T_AMB','GDI_35','DE_MES','D1','D2','F_MES','EU_SOL']
df2 = pd.DataFrame(zip(meses,dias,Tred,Tamb_mes,Gdi_35,DE_mes,D1_mes,D2_mes,f_mes,EU_solar),columns=columnas)
df2.head(12)

MES  DIAS  T_RED  T_AMB  GDI_35  DE_MES    D1    D2   F_MES  EU_SOL
0   ENE    31    8.4      6    2.55  207.82  0.55  2.86  0.3242   67.38
1   FEB    28    8.4      8    3.76  187.71  0.81  2.75  0.5190   97.42
2   MAR    31   10.4     11    4.42  199.76  1.00  2.87  0.6338  126.61
3   ABR    30   12.2     13    5.53  186.31  1.29  3.05  0.7844  146.14
4   MAY    31   14.2     18    5.69  184.46  1.39  3.05  0.8332  153.69
5   JUN    30   17.2     23    6.20  166.82  1.62  3.27  0.9221  153.82
6   JUL    31   20.2     28    7.08  160.30  1.99  3.53  1.0399  166.70
7   AGO    31   19.2     26    6.87  164.32  1.88  3.46  1.0081  165.65
8   SEP    30   17.2     21    5.67  166.82  1.48  3.44  0.8537  142.41
9   OCT    31   13.4     15    4.34  187.68  1.04  3.13  0.6435  120.77
10  NOV    30   10.4     11    3.06  193.32  0.69  2.87  0.4287   82.88
11  DIC    31    8.4      7    2.38  207.82  0.52  2.80  0.3040   63.18

Lo que dictamina el CTE DB-HE-4 de 2013 con respecto a la protección contra los sobrecalentamientos es que en ningún mes la generación solar debe superar en un 10% a la demanda energética estimada y que dicha generación no puede superar la demanda en tres meses seguidos. En nuestro caso, en julio la generación supera 4% la demanda y en agosto lo hace en un 0,8%. Por lo tanto, estamos dentro de las normas.

Antes de finalizar podemos calcular el rendimiento medio anual de la instalación, que resulta de sacar la proporción de la sumatoria anual de la energía solar útil entre el producto de la superficie de captación por la sumatoria anual de la energía incidente EI:

$\eta_{sistema} = {\sum EU_{solar} \over \sum S_C.EI_{mes}}$

In [38]:
print("Rendimiento anual del sistema: {:.1f}%".format(100*sum(EU_solar)/(2.4*sum(EI_mes))))

Rendimiento anual del sistema: 35.3%


El rendimiento medio anual del sistema es de **35,3%**. Es un poco bajo con respecto a la norma CTE DB-HE 4 2006 que busca un rendimiento igual o superior al 40%.

## Plan de vigilancia y plan de mantenimiento

Dependiendo de las instalaciones que haya que hacer en la vivienda para configurar este sistema de generación de ACS se tiene que adecuar un plan de vigilancia y mantenimiento siguiendo recomendaciones del CTE. Muchos de los componentes requieren revisión 